pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -1216,35 +1216,18 @@ func (c *Reconciler) updateLabelsAndAnnotations(ctx context.Context, pr *v1beta1
 	return newPr, nil
 }
 
-func storePipelineSpecAndMergeMeta(pr *v1beta1.PipelineRun, ps *v1beta1.PipelineSpec, meta *metav1.ObjectMeta) error {
+func storePipelineSpecAndMergeMeta(pr *v1beta1.PipelineRun, ps *v1beta1.PipelineSpec, meta *metav1.ObjectMeta, annotationsInStatus map[string]string) error {
 	// Only store the PipelineSpec once, if it has never been set before.
 	if pr.Status.PipelineSpec == nil {
 		pr.Status.PipelineSpec = ps
 
 		// Propagate labels from Pipeline to PipelineRun.
-		if pr.ObjectMeta.Labels == nil {
-			pr.ObjectMeta.Labels = make(map[string]string, len(meta.Labels)+1)
-		}
-		for key, value := range meta.Labels {
-			// Do not override duplicates between PipelineRun and Pipeline
-			// PipelineRun labels take precedences over Pipeline
-			if _, ok := pr.ObjectMeta.Labels[key]; !ok {
-				pr.ObjectMeta.Labels[key] = value
-			}
-		}
+		pr.ObjectMeta.Labels = taskrun.AppendMap(pr.ObjectMeta.Labels, meta.Labels)
 		pr.ObjectMeta.Labels[pipeline.PipelineLabelKey] = meta.Name
-
 		// Propagate annotations from Pipeline to PipelineRun.
-		if pr.ObjectMeta.Annotations == nil {
-			pr.ObjectMeta.Annotations = make(map[string]string, len(meta.Annotations))
-		}
-		for key, value := range meta.Annotations {
-			// Do not override duplicates between PipelineRun and Pipeline
-			// PipelineRun labels take precedences over Pipeline
-			if _, ok := pr.ObjectMeta.Annotations[key]; !ok {
-				pr.ObjectMeta.Annotations[key] = value
-			}
-		}
+		pr.ObjectMeta.Annotations = taskrun.AppendMap(pr.ObjectMeta.Annotations, meta.Annotations)
+		// Propagate annotations from resolutionrequest to PipelineRun's status.
+		pr.Status.Annotations = taskrun.AppendMap(pr.Status.Annotations, annotationsInStatus)
```

Comment:
I am thinking it might be better to do it after the ResolutionRequest supports the ConfigSource https://github.com/tektoncd/pipeline/issues/5529. 

Otherwise, we have to do `annotations from resolution` (map) -> `resolvedObjectMeta.ConfigSource` (struct) -> `annotations in run status` (map).

With https://github.com/tektoncd/pipeline/issues/5529 to be implemented, it makes it easier/cleaner. wdyt?

Reply:
+1 to returning a configSource vs a generic map

Reply:
Thank you for the feedback and idea @wlynch @dibyom !

With the structured fields supported in ResolutionRequest.Status and Run.Status, I've implemented the feedback so that the data is passed through those structured fields instead of annotation maps. The suggested `resolvedObjectMeta` is also implemented to pass the metadata together.

PTAL! Thank you very much!